nifi-commons/nifi-expression-language/src/test/java/org/apache/nifi/attribute/expression/language/TestQuery.java

```diff
@@ -846,7 +848,7 @@ public void testEscapeQuotes() {
         final String query = "startDateTime=\"${date:toNumber():toDate():format(\"" + format + "\")}\"";
         final String result = Query.evaluateExpressions(query, attributes, null);
 
-        final String expectedTime = new SimpleDateFormat(format, Locale.US).format(timestamp);
+        final String expectedTime = DateTimeFormatter.ofPattern(format, Locale.US).format(Instant.ofEpochMilli(timestamp).atZone(ZoneId.systemDefault()));
```

Comment:
In fact it was implicitly done before my change because `SimpleDateFormat.format` uses system timezone in this case but I agree with you that using system time zone in tests should be avoided and in production code should be well documented and test - fixed.